Feauture/add bugs repository

danielgomezrico opened the pull request:
Add `Send Bug Report on Shake` Switch to allow testers to upload bugs to pivotal with all phone info, log files, screenshot of the current page, custom title and description from the app faster.

Related to https://github.com/BaristaVentures/debug-artist/issues/4

danielgomezrico committed:
Add shake detector for report on shake option

danielgomezrico committed:
Use interface instead of implemenations for Shakes

danielgomezrico committed:
Add shake listener 	management

danielgomezrico committed:
Add pivotal repository and static upload

danielgomezrico committed:
Add properties to bug description

danielgomezrico committed:
Add error management

danielgomezrico committed:
Add bug report activity

danielgomezrico committed:
Fix with shake is enabled on start listener do nothing

danielgomezrico committed:
Use title and description from view to create bugs

danielgomezrico committed:
Add screenshot/upload management

danielgomezrico committed:
Add device logs to bug attachments

danielgomezrico committed:
Clean log picker management

danielgomezrico committed:
Refactor  BugReportRepository name to BugRepository

danielgomezrico committed:
Remove repeated read logs permission

danielgomezrico committed:
Fix tests

orendon commented on lib/src/main/java/com/barista_v/debug_artist/report_bug/pivotal/PrivotalService.kt:
```diff
@@ -0,0 +1,24 @@
+package com.barista_v.debug_artist.report_bug.pivotal
```
Privotal -> Pivotal

orendon commented on LICENSE:
```diff
@@ -177,7 +177,7 @@
    APPENDIX: How to apply the Apache License to your work.
 
-      To apply the Apache License to your work, attach the following
+      To apply the Apache License to your work, start the following
```
as far as I know Apache License 2.0 have not changed this

orendon commented on sample/src/debug/java/com/barista_v/sample/MyActivity.java:
```diff
@@ -26,11 +31,16 @@ public void setContentView(@LayoutRes int layoutResID) {
     String[] hosts = new String[] { "Value 1", "Value 2" };
 
+    BugRepository.Builder repositoryBuilder =
+        new BugReportRepositoryBuilder("8d4f9d99c2a4c818746fe1cb9015e2c9", "1954523",
```
any chance we don't expose the apikey and projectid in the codebase? maybe some configuration file?

orendon reviewed (commented):
Looking good

danielgomezrico committed:
Fix LICENSE had wrong text

danielgomezrico committed:
Fix typo on pivotal service file name

danielgomezrico committed:
Move keys to gradle.properties on sample

danielgomezrico committed:
Fix failing tests and now mock from interfaces instead of classes

danielgomezrico committed:
Refactor project structure

Move some classes to other packages to improve management

danielgomezrico committed:
Add mockito setup to mock final clasess / member methods

- Add mocking extensions to mock non final classes

danielgomezrico committed:
Remove unneeded open from debug drawer presenter

danielgomezrico committed:
Fix env variables script was is not called

danielgomezrico committed:
Add java options to define circleci memory

danielgomezrico committed:
Fix bintray properties creation broken on last commit

danielgomezrico committed:
Downgrade gradle 2.13 and android plugin 2.1.0

- Newer versions have issues with jacoco

danielgomezrico committed:
Downgrade gradle 2.13 and android plugin 2.1.0

- Newer versions have issues with jacoco

danielgomezrico committed:
Use string resource for report bug activity label

danielgomezrico committed:
Fix test dont create results files

- There was missing a @RunWIth at the top of each test.

danielgomezrico committed:
Clean Retry after timeout management

danielgomezrico committed:
Update report bug style

danielgomezrico committed:
Close report activity after success

danielgomezrico committed:
Add assemble step to circle

danielgomezrico committed:
Fix shake are not detected on some devices

danielgomezrico committed:
Update ... style on strings

danielgomezrico committed:
Add Jacobo rules to proguard setup

danielgomezrico committed:
Add empty fields validation

danielgomezrico committed:
Add ime action to description field

danielgomezrico committed:
Launch report activity as singleInstance

danielgomezrico committed:
Fix previews for pivotal tracker image attach

danielgomezrico committed:
Optimize image resources

danielgomezrico committed:
Update sample with image

danielgomezrico committed:
Add scroll view and new colors to report view

danielgomezrico committed:
Use 10 retries for timeout (pivotal have a lot of them)

danielgomezrico committed:
Add custom labels for bugs

danielgomezrico committed:
Fix shake disabled on start still opens report activity on shake

danielgomezrico committed:
Refactor package system

- Move pivotal tracker reporter to different module
- Rename main module to debugartist.menu.

danielgomezrico committed:
Fix circle ci setup
